### PR TITLE
Document alternatives to nested `ifelse()` 

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
Based on the feedback I have received from repos where I implemented lint workflow, I have come to realize that the short description we have here that mentions possible alternatives is not enough. It might be worth including some examples in the documentation itself.

I am especially not clear on what the proposed `{base}`-only alternative is supposed to look like. I've mentioned one of the many possibilities I could think of, but happy to change it based on feedback.

### CODE REVIEW

**Comment by Bisaloo** [comment]
Is there any reason to not use `switch()`?

```r
switch(
  x,
  "a" = 2L,
  "b" = 3L,
  1L
)
```

**Comment by IndrajeetPatil** [comment]
With `switch()`, you are restricted to a single expression, while the look-up table can deal with multiple expressions.

You _can_ also do that with `switch()`, but the solutions tend to be hacky. 

For example, covert booleans to strings and then use `switch()`:

```r
switch(paste(stability[x] == "sufficient", convergence[x] == "converged"),
  "TRUE TRUE" = paste0(
    text, " and the indices are reliable (ESS = ",
    insight::format_value(diagnostic$ESS, digits = 0), ")"
  ),
  "TRUE FALSE" = paste0(
    text, " even though the indices appear as reliable (ESS = ",
    insight::format_value(diagnostic$ESS, digits = 0), ")"
  ),
  "FALSE TRUE" = paste0(
    text, " but the indices are unreliable (ESS = ",
    insight::format_value(diagnostic$ESS, digits = 0), ")"
  ),
  "FALSE FALSE" = paste0(
    text, " and the indices are unreliable (ESS = ",
    insight::format_value(diagnostic$ESS, digits = 0), ")"
  )
)
```

A look-up table or a data frame provides the most general solution because you can easily use multiple expressions.

```r
dplyr::filter(df, stability == "sufficient", convergence == "converged")
```

Does that make sense?

**Comment by IndrajeetPatil** [comment]
I think we can mention both alternative `switch()` and look-up table, and qualify that with generalization available for each option.

**Comment by MichaelChirico** [comment]
My own experience is the difficulty of making the behavior work w.r.t missing inputs. It can be hard to keep track of which inputs being missing leads to missing output, and converting that to case expressions. Compounded by the behavior w.r.t. missingness is not often well-tested so it's easy to make a breaking change by accident.

File: R/nested_ifelse_linter.R [hunk 1]
```diff
@@ -9,9 +9,52 @@
...
+#'
+#'     ```r
+#'     data.table::fcase(
+#'       x == "a", 2L,
+#'       x == "b", 3L,
+#'       default = 1L
+#'     )
+#'     ```
+#'
+#'   - Use `dplyr::case_when()`
```
Comment (AshesITR) [inline R/nested_ifelse_linter.R:36]: This particular case should actually be handled by `dplyr::recode(x, "a" = 2L,  "b" = 3L, .default = 1L)`

Comment (IndrajeetPatil) [reply R/nested_ifelse_linter.R:36]: `recode()` has been superseded in favour of `case_match()`: https://dplyr.tidyverse.org/reference/recode.html

Comment (AshesITR) [reply R/nested_ifelse_linter.R:36]: Oh, thanks for the info. Then with `case_match()` accordingly.

File: R/nested_ifelse_linter.R [hunk 2]
```diff
@@ -9,9 +9,53 @@
...
```
Comment (AshesITR) [inline R/nested_ifelse_linter.R:36]: NB this is mismatched to the code.

Comment (IndrajeetPatil) [reply R/nested_ifelse_linter.R]: Good catch! Thanks. Fixed it.

**Review by AshesITR — APPROVED** [review]

**Pull request merged**